coralproject/talk
# Revamp comment details and implement toxicity detail

## What does this PR do?

- Refactored `ActionButton` into `ApproveButton` and `RejectButton`.
- Got rid of `utils/moderationQueueActionsMap.js` (finally..)
- Components `IfSlotIsEmpty` and `IfSlotIsNotEmpty` now accepts an array of slot names.
- HOC `excludeIf` now returns a new component.
- Introduce new Slots: `adminCommentMoreDetails`, `adminCommentMoreFlagDetails` that are only visible when user clicks on _more details_.
- Refactor flag details into new default plugin `talk-plugin-flag-details`.
- More flag details now only show user flag details.
- Implement toxic flag detail using new slot `adminCommentMoreFlagDetails` in `talk-plugin-toxic-comments`.
- Implement toxic detail using slot `adminCommentMoreDetails`.

https://www.pivotaltracker.com/story/show/150708082

## How do I test this PR?

-  Write toxic and non toxic-comments
- Flag some comments
- Check comment details area in mod queue and user details view.
- Non toxic comment should show toxic details
- Toxic comment should also show toxic details
- Comments with user flags should show more details about the users who created the flag
